platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/RosterUtils.java

```diff
@@ -193,6 +198,37 @@ public static RosterEntry getRosterEntry(@NonNull final Roster roster, final lon
         throw new RosterEntryNotFoundException("No RosterEntry with nodeId: " + nodeId + " in Roster: " + roster);
     }
 
+    /**
+     * Build an instance of RosterHistory from the current/previous AddressBook found in the PlatformState.
+     * @deprecated To be removed once AddressBook to Roster refactoring is complete.
+     * @param readablePlatformState
+     * @return a RosterHistory
+     */
+    @Deprecated(forRemoval = true)
+    @NonNull
+    public static RosterHistory buildRosterHistory(final PlatformStateAccessor readablePlatformState) {
+        if (readablePlatformState.getAddressBook() == null) {
+            throw new IllegalStateException("Address book is null");
+        }
+
+        final List<RoundRosterPair> roundRosterPairList = new ArrayList<>();
+        final Map<Bytes, Roster> rosterMap = new HashMap<>();
+
+        final Roster currentRoster = RosterRetriever.buildRoster(readablePlatformState.getAddressBook());
+        final Bytes currentHash = RosterUtils.hash(currentRoster).getBytes();
+        roundRosterPairList.add(new RoundRosterPair(readablePlatformState.getRound(), currentHash));
+        rosterMap.put(currentHash, currentRoster);
+
+        if (readablePlatformState.getPreviousAddressBook() != null) {
+            final Roster previousRoster = RosterRetriever.buildRoster(readablePlatformState.getPreviousAddressBook());
+            final Bytes previousHash = RosterUtils.hash(previousRoster).getBytes();
+            roundRosterPairList.add(new RoundRosterPair(0, previousHash));
```

Comment:
Just curious, why assume the previous address book should be paired with round 0?

Reply:
The short answer is because currently, there's no information in the PlatformState or elsewhere to precisely identify the round number since which the previous AddressBook has first become active.

A long and comprehensive answer is at https://github.com/hashgraph/hedera-services/pull/16067#discussion_r1829686955

Reply:
Thank you!